doc/source/advanced.md

```diff
@@ -199,3 +199,50 @@ To explicitly specify this strategy, use the following in your `config.yaml`:
 singleuser:
     schedulerStrategy: pack
 ```
+
+## Pre-pulling Images for Faster Startup
+
+When user images are of non-trivial sizes (as most of them are), **pre-pulling**
+the images on all nodes significantly decreases the amount of time it takes to
+start a user's server. Without pre-pulling, sometimes it can take upto 5-10
+minutes for a user's server to launch. With pre-pulling, it is cut down to a few
+seconds in most cases.
+
+By default, the user container image is pulled on to all nodes whenever a `helm
+install` or `helm upgrade` is performed. This causes `helm install` and `helm
+upgrade` to take several minutes, but mostly that is worth it since it makes
+user startup performance much faster. By default `helm install` or `helm
+upgrade` will wait for 5minutes before giving up. You can configure this by
+passing the `--wait <seconds>` flag to either of them.
+
+To disable the pre-pulling during `helm install` and `helm upgrade`, you can use
```

Comment:
We recommend using pre-pulling. For the rare cases where you have a good reason to disable it, pre-pulling can be disabled. To disable the pre-pulling during `helm install` and `helm upgrade`, you can use...


~~This is not recommended...it!~~